apps/login/src/app/api/loginname/route.ts

```diff
@@ -37,6 +46,91 @@ export async function POST(request: NextRequest) {
             return NextResponse.json(error, { status: 500 });
           });
       } else {
+        const loginSettings = await getLoginSettings(organization);
+        // TODO: check if allowDomainDiscovery has to be allowed too, to redirect to the register page
+        // user not found, check if register is enabled on organization
+
+        if (
+          loginSettings?.allowRegister &&
+          !loginSettings?.allowUsernamePassword
+        ) {
+          // TODO redirect to loginname page with idp hint
+          const identityProviders = await getActiveIdentityProviders(
+            organization,
+          ).then((resp) => {
+            return resp.identityProviders;
+          });
+
+          if (identityProviders.length === 1) {
+            const host = request.nextUrl.origin;
+
```

Comment:
Why not
```ts
if (identityProviders.length == 0) {
  // return early instead of putting everything into deep if branches like spaghetti
            return NextResponse.json(
              { message: "Could not find user" },
              { status: 404 },
            );
}

Reply:
I prefer it the other way, it feels more natural to me